crates/ruff_linter/src/checkers/ast/mod.rs

```diff
@@ -2576,7 +2576,7 @@ impl<'a> Checker<'a> {
                 } else {
                     if self.semantic.global_scope().uses_star_imports() {
                         if self.enabled(Rule::UndefinedLocalWithImportStarUsage) {
-                            self.diagnostics.push(
+                            self.diagnostics.get_mut().push(
```

Comment:
Nit: We could add a `report_diagnostic_mut` method to discourage direct `self.diagnostics` accessors (I'm afraid from people copying this code over using `self.report_diagnostic`)

Reply:
Oh yes, please!